pkg/query-service/utils/format.go

```diff
@@ -14,6 +14,8 @@ import (
 
 // ValidateAndCastValue validates and casts the value of a key to the corresponding data type of the key
 func ValidateAndCastValue(v interface{}, dataType v3.AttributeKeyDataType) (interface{}, error) {
+	// get the actual value if it's a pointer
+	v = getPointerValue(v)
```

Comment:
Why is this added?

Reply:
Added it due to this, so in the new list api that we are using, we add the id right, which I get as a pointer to a string.
https://github.com/SigNoz/signoz/blob/3fdfb51e0243d4c3587204c65152d71b9b2ea0df/pkg/query-service/app/querier/querier.go#L517
 
![image](https://github.com/user-attachments/assets/ad3b2710-0cfe-499e-8dfe-c77ebf1a270c)